internal/discord/commands.go

```diff
@@ -103,7 +103,93 @@ func (d *Discord) Purge(s *discordgo.Session, i *discordgo.InteractionCreate) {
 //	duration:	string
 //	reason:		string
 func (d *Discord) Exile(s *discordgo.Session, i *discordgo.InteractionCreate) {
-	return
+	optionMap := mapOptions(i)
+	logMsg, _ := d.LogCommand(i.Interaction)
+
+	state := &InteractionState{
+		session:     s,
+		interaction: i,
+		logMsg:      logMsg,
+		isFirst:     true,
+	}
+
+	var duration time.Duration
+	var err error
+
+	// Calculate duration of exile if argument is not empty
+	startTime := time.Now()
+	durationArg, hasDuration := optionMap["duration"]
+	if hasDuration {
+		duration, err = CalculateDuration(state, startTime, durationArg.StringValue())
+		if err != nil {
+			return
+		}
+	}
+
+	userToExile := optionMap["user"].UserValue(nil)
+
+	err = d.ExileUser(state, userToExile.ID, optionMap["reason"].StringValue())
+	if err != nil {
+		return
+	}
```

Comment:
it's unsettling have this in between two `if hasDuration` blocks.... do we have to calculate the duration before calling `ExileUser`? i think we would rather have it be together in the if block below

Reply:
I thought about that, but the issue is if the invoker formatted the duration argument incorrectly it would exile the user first then fail before carrying anything else out.

Reply:
I feel like there is a better way around this 🤔 lets open a new tech debt issue to deal with this and get this code merged in